coderd/coderdtest/swaggerparser.go

```diff
@@ -0,0 +1,328 @@
+		nodes, err := parser.ParseDir(fileSet, dir, nil, parser.ParseComments)
+		if err != nil {
+			return nil, xerrors.Errorf(`parser.ParseDir failed for "%s": %w`, dir, err)
+		}
+
+		for _, node := range nodes {
+			ast.Inspect(node, func(n ast.Node) bool {
+				commentGroup, ok := n.(*ast.CommentGroup)
+				if !ok {
+					return true
+				}
+
+				var isSwaggerComment bool
+				for _, line := range commentGroup.List {
+					text := strings.TrimSpace(line.Text)
+					if strings.HasPrefix(text, "//") && strings.Contains(text, "@Router") {
+						isSwaggerComment = true
+						break
+					}
+				}
+
+				if isSwaggerComment {
+					swaggerComments = append(swaggerComments, parseSwaggerComment(commentGroup))
+				}
+				return true
+			})
+		}
+	}
+	return swaggerComments, nil
+}
+
+func parseSwaggerComment(commentGroup *ast.CommentGroup) SwaggerComment {
+	c := SwaggerComment{
+		raw:        commentGroup.List,
+		parameters: []parameter{},
+		successes:  []response{},
+		failures:   []response{},
+	}
+	for _, line := range commentGroup.List {
+		splitN := strings.SplitN(strings.TrimSpace(line.Text), " ", 2)
+		if len(splitN) < 2 {
+			continue // comment prefix without any content
+		}
+		text := splitN[1] // Skip the comment prefix (double-slash)
+
+		if strings.HasPrefix(text, "@Router ") {
+			args := strings.SplitN(text, " ", 3)
+			c.router = args[1]
+			c.method = args[2][1 : len(args[2])-1]
+		} else if strings.HasPrefix(text, "@Summary ") {
+			args := strings.SplitN(text, " ", 2)
+			c.summary = args[1]
+		} else if strings.HasPrefix(text, "@ID ") {
+			args := strings.SplitN(text, " ", 2)
+			c.id = args[1]
+		} else if strings.HasPrefix(text, "@Success ") {
+			args := strings.Split(text, " ")
+
+			var success response
+			if len(args) > 1 {
+				success.status = args[1]
+			}
+			if len(args) > 2 {
+				success.kind = args[2]
+			}
+			if len(args) > 3 {
+				success.model = args[3]
+			}
+			c.successes = append(c.successes, success)
+		} else if strings.HasPrefix(text, "@Failure ") {
+			args := strings.Split(text, " ")
+
+			var failure response
+			if len(args) > 1 {
+				failure.status = args[1]
+			}
+			if len(args) > 2 {
+				failure.kind = args[2]
+			}
+			if len(args) > 3 {
+				failure.model = args[3]
+			}
+			c.failures = append(c.successes, failure)
+		} else if strings.HasPrefix(text, "@Tags ") {
+			args := strings.SplitN(text, " ", 2)
+			c.tags = args[1]
+		} else if strings.HasPrefix(text, "@Security ") {
+			args := strings.SplitN(text, " ", 2)
+			c.security = args[1]
+		} else if strings.HasPrefix(text, "@Param ") {
+			args := strings.SplitN(text, " ", 4)
+			p := parameter{
+				name: args[1],
+				kind: args[2],
+			}
+			c.parameters = append(c.parameters, p)
+		} else if strings.HasPrefix(text, "@Accept ") {
+			args := strings.SplitN(text, " ", 2)
+			c.accept = args[1]
+		} else if strings.HasPrefix(text, "@Produce ") {
+			args := strings.SplitN(text, " ", 2)
+			c.produce = args[1]
+		}
```

Comment:
Would this be better refactored as a switch statement?

Also, what do we want to do in the case of a prefix that doesn't match any of the above?

Reply:
Sure, I can refactor it, maybe I can squeeze it a bit.

> Also, what do we want to do in the case of a prefix that doesn't match any of the above?

We ignore it as it means that's a property that the validator does not process.

Reply:
> We ignore it as it means that's a property that the validator does not process.
👍